extensions/cloud/portability-cloud-google/src/main/java/org/dataportabilityproject/cloud/google/GoogleCloudExtension.java

```diff
@@ -0,0 +1,89 @@
+package org.dataportabilityproject.cloud.google;
+
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.auth.oauth2.GoogleCredentials;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.common.base.Preconditions;
+import com.google.inject.Inject;
+import org.dataportabilityproject.api.launcher.ExtensionContext;
+import org.dataportabilityproject.spi.cloud.extension.CloudExtension;
+import org.dataportabilityproject.spi.cloud.storage.BucketStore;
+import org.dataportabilityproject.spi.cloud.storage.CryptoKeyStore;
+import org.dataportabilityproject.spi.cloud.storage.JobStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GoogleCloudExtension implements CloudExtension {
+  // TODO: Should these be initialized from the ExtensionContext and not injected?
+  // If injected, this requires that we install the GoogleCloudModule before using which might not
+  // make sense depending on where it would happen
+  private final GoogleCredentials googleCredentials;
+  private final String projectId;
+  private final HttpTransport transport;
+  private final JsonFactory jsonFactory;
+  private static final Logger logger = LoggerFactory.getLogger(GoogleCloudExtension.class);
+
+  private boolean initialized = false;
+  private Datastore datastore;
+  private JobStore jobstore;
+  private BucketStore bucketStore;
+  private GoogleCryptoKeyStore cryptoKeyManagementSystem;
+
+  @Inject
+  GoogleCloudExtension(
```

Comment:
I think extensions will have no constructor if I understand correctly?

Reply:
I think just a public no-arg one right? I think that's required by the ServiceLoader

Reply:
Changed to have no constructor. I'm not convinced I'm creating the transport and the json factory correctly, but thats how it was done throughout the project - I don't have anything runnable to test it with though